docs/book/src/architecture/controllers/cluster.md

```diff
@@ -1 +1,45 @@
-# Cluster
+# Cluster Controller
+
+The cluster controller is responsible for keeping the cluster object in sync with the infrastructure cluster object.
+
+Its main responsibilities are:
+
+* Setting an OwnerReference on the infrastructure object.
+* Deleting all related objects in the correct order.
+* Keeping the Cluster object up to date with external objects.
+* Creating a kubeconfig secret for child clusters.
+
+## Contracts
+
+### Infrastructure Provider
+
+The InfrastructureCluster object **must** have a status field.
+
+In the status field it must have:
+
+* `ready` - a boolean field
+* `apiEndpoints` - a slice of strings
```

Comment:
I'd copy the definition we had in the bootstrap proposal, which had a little more details

Reply:
I've added more detail instead as these are not outlined in the bootstrap proposal, assuming you mean https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20190610-machine-states-preboot-bootstrapping.md.